spec/lib/psulib_traject/processors/summary_holdings_spec.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+RSpec.describe PsulibTraject::Processors::SummaryHoldings do
+  describe '::call' do
+    subject(:summary_holdings) { described_class.call(record: record) }
+
+    context 'when there are no summary holdings' do
+      let(:record) { MarcBot.build(:record) }
+
+      it { is_expected.to be_empty }
+    end
+
+    context 'when there is a summary holdings unit' do
+      let(:record) do
+        MarcBot.build(:record,
+                      f_852: { b: 'UP-SPECCOL', c: 'HC-SERIALS', h: 'HD6515.P4 I5 F' },
+                      f_866: 'v.12(1956/57)-v.40:1-7/8(1984)',
+                      f_867: 'Supplemental',
+                      f_868: 'Index')
+      end
+
+      it 'returns a one Unit object' do
+        holdings = summary_holdings.first
+        expect(holdings).to be_a(PsulibTraject::PeriodicalHoldings)
+        expect(holdings.library).to eq('UP-SPECCOL')
+        expect(holdings.location).to eq('HC-SERIALS')
+        expect(holdings.summary).to contain_exactly('v.12(1956/57)-v.40:1-7/8(1984)')
+        expect(holdings.supplement).to contain_exactly('Supplemental')
+        expect(holdings.index).to contain_exactly('Index')
+      end
+    end
+
+    context 'when multiple summary holdings are present' do
+      let(:record) { MARC::Reader.new(File.join('./spec/fixtures/summary_holdings.mrc')).to_a.first }
+
+      its(:count) { is_expected.to eq(3) }
+    end
+  end
```

Comment:
@awead Could we also test for when there are multiple 852s? Another potential one maybe to test when there is a 852 with no summary followed by a 852 with summary?

Reply:
Sure thing!